### ISSUE TITLE
Two accessibility issues for screen reader users

### ISSUE THREAD

**Issue by Almost-Senseless-Coder** [welpo/tabi#182]
# Bug Report

## Environment

Zola version: 0.17.2
tabi commit: TBD

## Expected Behavior

- The light and dark mode toggle should be represented as a toggle button to screen reader users.
- The copy code to clipboard button should be represented to screen reader users as a button and should report copy action completion.

## Current Behavior

Both buttons are currently invisible and unclickable for screen reader users.

## Step to Reproduce

Turn any screen reader of your choosing on (such as Orca on Linux, NVDA on Windows, TalkBack on Android or VoiceOver on Mac OS), navigate to a Zola-egenerated website with Tabi, and look for the dark/light mode toggle. You won't find it. Likewise, you won't find the ccopy to clipboard button.

I have already fixed that issue for my website, but it currently breaks internationalisation as the JS provides text in English. Please help me find a better solution. (See <https://codeberg.org/Tim-Boettcher/tim-boettcher-online/src/branch/main/static/js>).

**Comment by Almost-Senseless-Coder** [comment]
I can't seem to figure out How to edit the issue description, but here
S the tabi commit: 9c9ba7f73478cd64984e8917d8cfbcf0be2acd04

**Comment by welpo** [comment]
Hey Tim!

Thanks a lot for bringing this to my attention.

I see the language picker is also unavailable for screen reader users.

I am working on a fix for all 4 issues:
- Light/dark theme toggle
- Copy to clipboard
- Internationalisation for above
- Unavailable language picker

**Comment by Almost-Senseless-Coder** [comment]
Let me know if you need any help with anything.

**Comment by welpo** [comment]
Regarding the theme switcher, I see you've kept the `title` "Toggle dark/light mode" and used the `aria-label` "Toggle dark mode".

Do you think it would be appropriate for the `aria-label` to match the title (i.e. "Toggle dark/light theme")?

Or perhaps it should switch based on the current theme?

**Comment by Almost-Senseless-Coder** [comment]
This is a matter of semantics.

When `role="button"` gets combined with `aria-pressed="true|false"`, it creates a toggle button element for screen readers, which reflect two states: on and off (somewhat similar to radio buttons, but radio bu`tons usually occur in groups). If we call it "toggle dark/light mode", then screen reader users don't know which state maps to which theme mode.

I opted for calling it "Toggle dark mode" for blind users, because then it's clear: If the toggle is on, dark mode is active. I left the title be because I wanted to make minimal changes to the JS and for a computer mouse user that title should be fine, since they can see which mode is currently active.

I don't think swapping labels based on state makes much sense for a toggle button, since there's already an on/ofc indicatoo. But you could use a plain button (without the `aria-pressed` attoibute) and swap labels to indicate what clicking the button will do.

**Comment by welpo** [comment]
Ah, that makes sense. Thank you.

Regarding the language switcher, I've used a `details`/`summary` structure. Now the screen reader is able to detect it and interact with it.

Do you think this implementation is effective, in terms of labels ("Language selection") and structure?:

```tera
    <details class="dropdown">
        <summary role="button" aria-haspopup="true" aria-label="{{ macros_translate::translate(key="language_selection", default="Language selection", language_strings=language_strings) }}">
            <div class="language-switcher-icon"></div>
        </summary>
        <div class="dropdown-content" role="menu">
```

**Comment by Almost-Senseless-Coder** [comment]
I think that should do the trick, yeah. 

Tip: Add `lang="<language identifier>"` to each of the language options (as in, the menu items). That results in the correct scoeen reader language getting used to pronounce the language name.

**Comment by welpo** [comment]
Thanks for the tip and the guidance.

I believe I've solved all issues discussed in PR #183.

Please do let me know if you encounter any other issues, and thanks for using tabi!
